blivet/osinstall.py

```diff
@@ -71,6 +151,50 @@ def copy_to_system(source):
     return True
 
 
+def update_from_anaconda_flags(blivet_flags, anaconda_flags):
+    """
+    Set installer-specific flags. This changes blivet default flags by
+    either flipping the original value, or it assigns the flag value
+    based on anaconda settings that are passed in.
+
+    :param blivet_flags: Blivet flags
+    :type flags: :class:`blivet.flags.Flags`
+    :param anaconda_flags: anaconda flags
+    :type anaconda_flags: :class:`pyanaconda.flags.Flags`
+    """
+    # always enable the debug mode when in the installer mode so that we
+    # have more data in the logs for rare cases that are hard to reproduce
+    blivet_flags.debug = True
+    blivet_flags.testing = anaconda_flags.testing
+    blivet_flags.automated_install = anaconda_flags.automatedInstall
+    blivet_flags.live_install = anaconda_flags.livecdInstall
+    blivet_flags.image_install = anaconda_flags.imageInstall
+
+    blivet_flags.selinux = anaconda_flags.selinux
+
+    blivet_flags.gfs2 = "gfs2" in blivet_flags.boot_cmdline
+    blivet_flags.jfs = "jfs" in blivet_flags.boot_cmdline
+    blivet_flags.reiserfs = "reiserfs" in blivet_flags.boot_cmdline
+
+    blivet_flags.arm_platform = anaconda_flags.armPlatform
+    blivet_flags.gpt = anaconda_flags.gpt
+
+    blivet_flags.multipath_friendly_names = anaconda_flags.mpathFriendlyNames
+    blivet_flags.allow_imperfect_devices = anaconda_flags.rescue_mode
+
+    blivet_flags.ibft = anaconda_flags.ibft
+    blivet_flags.dmraid = anaconda_flags.dmraid
+
+    # We don't want image installs writing backups of the *image* metadata
+    # into the *host's* /etc/lvm. This can get real messy on build systems.
+    if blivet_flags.image_install:
+        blivet_flags.lvm_metadata_backup = False
+
+    blivet_flags.auto_dev_updates = True
+    blivet_flags.selinux_reset_fcon = True
+    blivet_flags.keep_empty_ext_partitions = False
```

Comment:
Please move these flag settings that aren't conditional up into `enable_installer_mode`.